packages/dashboard/src/components/tasks/task-page.tsx

```diff
@@ -14,59 +13,58 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-/**
- * Sort tasks in place, by priority, then by start time.
- */
-function sortTasks(tasks: RmfModels.TaskSummary[]) {
-  tasks.sort((a, b) => {
-    const aPriority = a.task_profile.description.priority.value;
-    const bPriority = b.task_profile.description.priority.value;
-    if (aPriority === bPriority) {
-      const aStartTime = a.start_time.sec;
-      const bStartTime = b.start_time.sec;
-      return aStartTime - bStartTime;
-    }
-    return aPriority - bPriority;
-  });
-  return tasks;
-}
-
 export function TaskPage() {
   const classes = useStyles();
   const { tasksApi = null } = React.useContext(RmfIngressContext) || {};
   const places = React.useContext(PlacesContext);
-  const [taskSummaries, setTaskSummaries] = React.useState<RmfModels.TaskSummary[]>([]);
-
-  const handleRefresh = React.useCallback(async () => {
-    if (!tasksApi) {
-      return;
-    }
-    const tasks = await tasksApi.getTasksTasksGetTasksGet();
-    const getTaskSummaries = tasks.data.map((task: TaskProgress) => task.task_summary);
-    sortTasks(getTaskSummaries);
-    setTaskSummaries(getTaskSummaries);
-  }, [tasksApi]);
 
-  React.useEffect(() => {
-    handleRefresh();
-  }, [handleRefresh]);
+  const fetchTasks = React.useCallback<TaskPanelProps['fetchTasks']>(
+    async (limit: number, offset: number) => {
+      if (!tasksApi) {
+        return {
+          tasks: [],
+          totalCount: 0,
+        };
+      }
+      const resp = await tasksApi.getTasksTasksGet(
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        limit,
+        offset,
+        'priority,-start_time',
+      );
```

Comment:
Just to check is it possible to change `getTasksTasksGet` to an object? Or wrap it? 

e.g: await tasksApi.getTasksTasksGet({limit, offset, options: 'priority,-start_time'}

Reply:
unfortunately that is generated from swagger codegen, i havent found a way to change the args yet.